curriculum/challenges/english/14-responsive-web-design-22/learn-html-forms-by-building-a-registration-form/60fab8367d35de04e5cb7929.md

```diff
@@ -33,6 +33,14 @@ You should give the `a` element an `href` of `https://www.freecodecamp.org/news/
 assert.match(document.querySelector('fieldset:nth-child(3) + label > input + a')?.href, /https:\/\/www\.freecodecamp\.org\/news\/terms-of-service\/?/);
 ```
 
+You should not have extra spaces around the text in the `a` element.
+
+```js
+const nestedAnchor = document.querySelector('fieldset:nth-child(3) + label > input + a')?.textContent;
+const innerContent = nestedAnchor.innerHTML;
+assert.isNotTrue(/^\s+|\s+$/.test(innerContent));
+```
+
 You should only wrap the `a` element around the text `terms and conditions`.
```

Comment:
The new hint is supposed to make it clear to campers that they have white space around the text `terms and conditions` where before it was not immediately obvious that was the issue.

Reply:
Actually they aren't testing the same thing. 

The existing hint checks for the exact text "terms and conditions" and will only pass if the text is "terms and conditions". 

My hint will allow any text to go through, provided there are no spaces : "trms and coditions"  will pass. Then the next hint will be about the proper text.

Reply:
This approach was suggested [here](https://github.com/freeCodeCamp/freeCodeCamp/issues/51961#issuecomment-1765143257).

I quote: "I think the actual implementation details of the tests are better suited in a PR. I will open this up to contribution.

If other challenges need the same change we can collect them in this issue and make it more general." 

This is another challenge that needed the same change as the initial challenge I pointed out.

Reply:
I'd rather do the first and make it similar to those in that PR.

Reply:
I can't believe I forgot about this PR. Let's see if this finishes it. I reordered the tests so that the content one came before the one about the spacing.